src/test/java/org/opensearch/knn/index/OpenSearchIT.java

```diff
@@ -483,4 +486,151 @@ public void testIndexingVectorValidation_updateVectorWithNull() throws Exception
+        final String indexName = "test-index-with-build-graph-settings";
+        createKnnIndex(indexName, settings, knnIndexMapping);
+        final String buildVectorDataStructureThresholdSetting = getIndexSettingByName(
+            indexName,
+            KNNSettings.INDEX_KNN_BUILD_VECTOR_DATA_STRUCTURE_THRESHOLD
+        );
+        assertNotNull("build_vector_data_structure_threshold index setting is not found", buildVectorDataStructureThresholdSetting);
+        assertEquals(
+            "incorrect setting for build_vector_data_structure_threshold",
+            buildVectorDataStructureThreshold,
+            Integer.valueOf(buildVectorDataStructureThresholdSetting)
+        );
+        deleteKNNIndex(indexName);
+    }
+
+    public void testKNNIndex_whenBuildThresholdIsNotProvided_thenShouldNotReturnSetting() throws Exception {
+        final String knnIndexMapping = createKnnIndexMapping(FIELD_NAME, KNNEngine.getMaxDimensionByEngine(KNNEngine.DEFAULT));
+        final String indexName = "test-index-with-build-graph-settings";
+        createKnnIndex(indexName, knnIndexMapping);
+        final String buildVectorDataStructureThresholdSetting = getIndexSettingByName(
+            indexName,
+            KNNSettings.INDEX_KNN_BUILD_VECTOR_DATA_STRUCTURE_THRESHOLD
+        );
+        assertNull(
+            "build_vector_data_structure_threshold index setting should not be added in index setting",
+            buildVectorDataStructureThresholdSetting
+        );
+        deleteKNNIndex(indexName);
+    }
+
+    public void testKNNIndex_whenGetIndexSettingWithDefaultIsCalled_thenReturnDefaultBuildGraphThresholdValue() throws Exception {
+        final String knnIndexMapping = createKnnIndexMapping(FIELD_NAME, KNNEngine.getMaxDimensionByEngine(KNNEngine.DEFAULT));
+        final String indexName = "test-index-with-build-vector-graph-settings";
+        createKnnIndex(indexName, knnIndexMapping);
+        final String buildVectorDataStructureThresholdSetting = getIndexSettingByName(
+            indexName,
+            KNNSettings.INDEX_KNN_BUILD_VECTOR_DATA_STRUCTURE_THRESHOLD,
+            true
+        );
+        assertNotNull("build_vector_data_structure index setting is not found", buildVectorDataStructureThresholdSetting);
+        assertEquals(
+            "incorrect default setting for build_vector_data_structure_threshold",
+            KNNSettings.INDEX_KNN_DEFAULT_BUILD_VECTOR_DATA_STRUCTURE_THRESHOLD,
+            Integer.valueOf(buildVectorDataStructureThresholdSetting)
+        );
+        deleteKNNIndex(indexName);
+    }
+
+    /*
+        For this testcase, we will create index with setting build_vector_data_structure_threshold as -1, then index few documents, perform knn search,
+        then, confirm no hits since there are no graph. In next step, update setting to 0, force merge segment to 1, perform knn search and confirm expected
+        hits are returned.
+     */
```

Comment:
Can we also have a test right away which has a threshold > 0 and then see the behavior when we try to search

Reply:
Ack

Reply:
Added new test case